### PR TITLE
Feat/civ 2898 notification

### PR DESCRIPTION
**Pull request by bhagyashreesharma90**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-2898

### Change description ###
Changes for evidence upload scheduler

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/handler/tasks/EvidenceUploadCheckHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        when(mockTask.getWorkerId()).thenReturn("worker");
+    }
+
+    @Test
+    void shouldEmitEvidenceUploadCheckEvent_whenCasesFound() {
+        long caseId = 1L;
+        Map<String, Object> data = Map.of("data", "some data");
+        List<CaseDetails> caseDetails = List.of(CaseDetails.builder().id(caseId).data(data).build());
+
+        when(searchService.getCases()).thenReturn(caseDetails);
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/tasks/EvidenceUploadCheckHandlerTest.java:52]: Minor:
this should be a "given".

`given(searchService.getCases()).willReturn(caseDetails);`

In general, in each test, put three comments:
```// Given
[code for the "given" section of the test: that is, the assumptions around the test]
// When
[code for the "when" section of the test: that is, the call to the method you want to test]
// Then
[code for the "then" section of the test: that is, the assertions you want to verify]
```
In an ideal world we would do BDD tests but as this is not the chosen approach, we need to try and get the best out of the tools we use. Following this approach helps clarifying the structure of the tests and makes them more readable.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/tasks/EvidenceUploadCheckHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        verify(externalTaskService).complete(mockTask);
+    }
+
+    @Test
+    void shouldNotEmitEvidenceUploadCheckEvent_WhenNoCasesFound() {
+        when(searchService.getCases()).thenReturn(List.of());
+
+        handler.execute(mockTask, externalTaskService);
+
+        verifyNoInteractions(applicationEventPublisher);
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/tasks/EvidenceUploadCheckHandlerTest.java:64]: Same as above.

Try this:
```
// Given
given(searchService.getCases()).willReturn(List.of());

// When
handler.execute(mockTask, externalTaskService);

// Then
verifyNoInteractions(applicationEventPublisher);
```

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java [hunk 3]
```diff
@@ -123,6 +125,8 @@ public enum CaseEvent {
     NOTIFY_DIRECTION_ORDER_DJ_DEFENDANT(CAMUNDA),
     NOTIFY_APPLICANT_SOLICITOR1_FOR_CONTACT_DETAILS_CHANGE(CAMUNDA),
     NOTIFY_APPLICANT_SOLICITOR1_FOR_DEFENDANT_RESPONSE_CUI(CAMUNDA),
+    NOTIFY_RESPONDENT_SOLICITOR_FOR_EVIDENCE_UPLOAD(CAMUNDA),
+    NOTIFY_APPLICANT_SOLICITOR_FOR_EVIDENCE_UPLOAD(CAMUNDA),
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java:129]: Do we have any test around these new case events?

File: src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/EvidenceUploadNotificationExternalTaskListener.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+public class EvidenceUploadNotificationExternalTaskListener {
+
+    private static final String TOPIC = "EVIDENCE_UPLOAD_CHECK";
+
+    @Autowired
+    private EvidenceUploadNotificationExternalTaskListener(EvidenceUploadCheckHandler evidenceUploadCheckHandler,
+                                                           ExternalTaskClient client) {
+        TopicSubscriptionBuilder subscriptionBuilder = client.subscribe(TOPIC);
+        subscriptionBuilder.handler(evidenceUploadCheckHandler).open();
+    }
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/EvidenceUploadNotificationExternalTaskListener.java:19]: I can see why you didn't write a test around this: none of the other Task Listeners have a test.

I can also see that they are all basically the same and we could add some abstraction around them.

Please create a ticket in TechDebt, assign it to me and notify me, so I am aware of this repetition and the lack of tests and I can figure out a way to best do it. Once you created a ticket and notified me, mark this conversation as resolved.

**Review by m-meulendijks-v1 — COMMENTED** [review]
Is there anything in this PR that is not in the other one? If not, I would just show us the previous PR if that covers the entire journey.

**Comment by bhagyashreesharma90** [comment]
> Is there anything in this PR that is not in the other one? If not, I would just show us the previous PR if that covers the entire journey.

The PR https://github.com/hmcts/civil-service/pull/1722/ covers entire journey and has all changes from this PR also.

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java (hunk 3 above)
Comment (bhagyashreesharma90) [reply src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java:129]: EvidenceUploadApplicantNotificationHandler and EvidenceUploadRespondentNotificationHandler will cover this 
and also we have test in camunda to check this events..
Camunda PR https://github.com/hmcts/civil-camunda-bpmn-definition/pull/369

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java [hunk 5]
```diff
@@ -50,6 +50,8 @@ public enum CaseEvent {
     APPLY_NOC_DECISION(USER),
     EVIDENCE_UPLOAD(USER),
     migrateCase(USER),
+    EVIDENCE_UPLOAD_CHECK(USER),
+    EVIDENCE_UPLOAD_NOTIFICATION(USER),
```
[the remark this answers is not in the document]
Comment (bhagyashreesharma90) [reply src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java:54]: EvidenceUploadNotificationCallbackHandlerTest will cover for notification and we have test for these two events in camunda
Camunda PR https://github.com/hmcts/civil-camunda-bpmn-definition/pull/369

**Review by hmcts-version1-ignacio — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/EvidenceUploadNotificationExternalTaskListener.java (hunk 4 above)
Comment (bhagyashreesharma90) [reply src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/EvidenceUploadNotificationExternalTaskListener.java:19]: @filippo-possenti-version1 , Created ticket https://tools.hmcts.net/jira/browse/CIV-5794

**Pull request closed without merging**